presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -1,118 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.facebook.presto.sql.planner.optimizations;
-
-import com.facebook.presto.Session;
-import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
-import com.facebook.presto.sql.planner.SymbolAllocator;
-import com.facebook.presto.sql.planner.TypeProvider;
-import com.facebook.presto.sql.planner.plan.Assignments;
-import com.facebook.presto.sql.planner.plan.LateralJoinNode;
-import com.facebook.presto.sql.planner.plan.PlanNode;
-import com.facebook.presto.sql.planner.plan.ProjectNode;
-import com.facebook.presto.sql.planner.plan.SimplePlanRewriter;
-import com.facebook.presto.sql.planner.plan.ValuesNode;
-
-import java.util.List;
-
-import static com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
-import static com.facebook.presto.sql.planner.plan.SimplePlanRewriter.rewriteWith;
-import static java.util.Objects.requireNonNull;
-
-/**
- * This optimizer can rewrite correlated single row subquery to projection in a way described here:
- * From:
- * <pre>
- * - Lateral(with correlation list: [A, C])
- *   - (input) plan which produces symbols: [A, B, C]
- *   - (subquery)
- *     - Project (A + C)
- *       - single row VALUES()
- * </pre>
- * to:
- * <pre>
- *   - Project(A, B, C, A + C)
- *       - (input) plan which produces symbols: [A, B, C]
- * </pre>
- */
-public class TransformCorrelatedSingleRowSubqueryToProject
```

Comment:
Typically during migration we left the legacy rule for couple releases just in case of bug introduction during the migration.

Reply:
No, please revert this change.

Reply:
Ack, will do